### PR TITLE
Add simple test of "registry upload bulk protos"

### PR DESCRIPTION
**Pull request by timburks**
This tests `registry upload bulk protos` by uploading a directory containing descriptions of two Google APIs and confirming that two specs are uploaded and that each zipped spec includes the expected number of proto and metadata files.

### CODE REVIEW

File: cmd/registry/cmd/upload/bulk/protos.go [hunk 1]
```diff
@@ -372,7 +372,7 @@ func referencedProtos(protos []string, root string) ([]string, error) {
 		return nil, err
 	}
 	defer os.RemoveAll(tempDir)
-	args := []string{"-o", tempDir + "/proto.pb"}
+	args := []string{"-o", tempDir + "/proto.pb", "--include_imports"}
```
Comment (timburks) [inline cmd/registry/cmd/upload/bulk/protos.go:375]: Testing revealed that the "--include_imports" flag had been omitted in the original commit (which had relied on iteratively compiling a list of protos until all were included). Subsequent removal of that iteration broke the uploader.

File: cmd/registry/cmd/upload/bulk/protos.go [hunk 2]
```diff
@@ -395,11 +395,9 @@ func protosFromFileDescriptorSet(filename string) ([]string, error) {
...
 	filenameset := make(map[string]bool)
 	for _, file := range fds.File {
-		filenameset[*file.Name] = true
-		for _, dependency := range file.Dependency {
-			if !strings.HasPrefix(dependency, "google/protobuf/") {
-				filenameset[dependency] = true
-			}
+		filename = *file.Name
+		if !strings.HasPrefix(filename, "google/protobuf/") {
+			filenameset[filename] = true
```
Comment (timburks) [inline cmd/registry/cmd/upload/bulk/protos.go:400]: With `--include_imports`, we no longer need to go through the list of dependencies. All dependencies are included in the list of compiled files.

File: cmd/registry/cmd/upload/bulk/protos_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
```
Comment (seaneganx) [inline cmd/registry/cmd/upload/bulk/protos_test.go:1]: 2022

File: cmd/registry/cmd/upload/bulk/protos_test.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	args := []string{"protos", "testdata/protos", "--project-id", projectID}
+	cmd.SetArgs(args)
+	if err := cmd.Execute(); err != nil {
+		t.Fatalf("Execute() with args %+v returned error: %s", args, err)
+	}
+	tests := []struct {
+		desc                string
+		spec                string
+		want_proto_count    int
+		want_metadata_count int
```
Comment (seaneganx) [inline cmd/registry/cmd/upload/bulk/protos_test.go:86]: wantProtoCount, wantMetadataCount

**Review by seaneganx — APPROVED** [review]

File: cmd/registry/cmd/upload/bulk/protos_test.go (hunk 3 above)
Comment (timburks) [reply cmd/registry/cmd/upload/bulk/protos_test.go]: Thanks, fixed.

File: cmd/registry/cmd/upload/bulk/protos_test.go (hunk 4 above)
Comment (timburks) [reply cmd/registry/cmd/upload/bulk/protos_test.go]: Thanks, fixed.

**Pull request merged**